src/gazebo_motor_failure_plugin.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2017 Nuno Marques, PX4 Pro Dev Team, Lisbon
+ * Copyright 2017 Siddharth Patel, NTU Singapore
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "gazebo_motor_failure_plugin.h"
+
+namespace gazebo {
+
+GazeboMotorFailure::~GazeboMotorFailure() {
+  event::Events::DisconnectWorldUpdateBegin(updateConnection_);
+}
+
+
+void GazeboMotorFailure::Publish() {
+  motor_failure_msg_.set_data(motor_Failure_Number_);
+  motor_failure_pub_->Publish(motor_failure_msg_);
+}
+
+/// \brief Create subscription to ROS topic that triggers the motor failure
```

Comment:
This documentation needs to be on `gazebo_motor_failure_plugin.h` header file, in this case above https://github.com/PX4/sitl_gazebo/pull/140/files#diff-14fd62890b0e31ba98873f592f5ea91eR62